### ISSUE TITLE
Revisit keymap#35 - international keyboard support.

### ISSUE THREAD

**Issue by ccoenen** [atom/atom#9455]
International Support in atom is somewhat lackluster. In reality it's not, but it sure feels that way. If you live anywhere outside the US, it means you usually don't have a US Keyboard layout. This is pretty much everyone.

There has been a lot of tickets on this topic, and they're all closed in favor of this one: https://github.com/atom/atom-keymap/issues/35 my personal impression is, that it's too easy to ignore when you're on an US keyboard yourself. This is, why i'm raising the issue here again, and not in the keymap project.

I would like to petition for one of two things, both of which are not *too* hard to do. Both would make the experience for new atom users a lot better.

a) you could either make the popular (and apparently amazing) package `keyboard-localization` part of EVERY atom install automatically.

b) you could remove all Ctrl+Alt Shortcuts from the default shortcuts.

Either one will work around the issue.

The actual solution would be to correctly support the AtlGr Key, but this seems to be a problem upstream, and there's no telling when that can get resolved.

**Comment by jeancroy** [comment]
In addition to (a), the [keyboard-layout](https://github.com/atom/keyboard-layout) package should probably support other platform than OS X. And then we could do something, at the minimum warn the user, at best enable the keyboard-localization package for the proper layout.

This is because when a key doesn't work, say `[` the first reflex is not: let's see if there's a package I can configure.

**Comment by benogle** [comment]
cc @nathansobo. IIRC, we are waiting on something landing on chrome. Know that this is definitely on our short list.

**Comment by ccoenen** [comment]
These two chromium issues seem related, but i do not know enough about atom's internals to be certain:

* https://code.google.com/p/chromium/issues/detail?id=263724
* https://code.google.com/p/chromium/issues/detail?id=227231

**Comment by jeancroy** [comment]
Also bracket have an environment similar to atom ?
They seems able to distinguish between ctrl+alt vs rigth alt and disable keybinding in the second case.
https://github.com/adobe/brackets/pull/10109

**Comment by nathansobo** [comment]
Our Summer of Code student @pandrei [researched this](https://github.com/atom/gsoc-atom/blob/master/atom_layout.md) a few months ago. From my inspection this weekend, it looks like support for `.key` and `.code` on keyboard events is getting close but isn't quite there yet.

I'm reluctant to spend resources on a hack when a true fix at the browser level is impending.

**Comment by lee-dohm** [comment]
This is a duplicate of atom/atom-keymap#35. Interested people should subscribe there for updates.

We're trying to control the amount of Issues that we have to triage and track](http://blog.atom.io/2016/04/19/managing-the-deluge-of-atom-issues.html). In the future, please continue discussion on the original issue rather than opening an intentional duplicate.

**Comment by ccoenen** [comment]
@lee-dohm It's not really a duplicate. I'm advocating for a workaround for the short-term until the long-term fix is implemented. People have been working around thiss issues since pre-1.0. It's affecting pretty much EVERYONE who's not on an US-layout keyboard.
